fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -244,10 +245,16 @@ private FedoraResource getBaseVersion(final FedoraResource resource) {
             .filter(checkAccessTo.or(checkAccessToClass))
             .forEach(auth -> {
                 concat(auth.getAgents().stream(), dereferenceAgentGroups(auth.getAgentGroups()).stream())
+                    .filter(agent -> !agent.equals(FOAF_AGENT_VALUE))
                     .forEach(agent -> {
                         effectiveRoles.computeIfAbsent(agent, key -> new HashSet<>())
                             .addAll(auth.getModes().stream().map(URI::toString).collect(toSet()));
                     });
+                auth.getAgentClasses().stream().filter(agentClass -> agentClass.equals(FOAF_AGENT_VALUE))
+                    .forEach(agentClass -> {
+                        effectiveRoles.computeIfAbsent(agentClass, key -> new HashSet<>())
+                            .addAll(auth.getModes().stream().map(URI::toString).collect(toSet()));
+                    });
```

Comment:
@mohideen it looks like you filter out `FOAF_AGENT_VALUE` from `effectiveRoles` in line 248 and then re-add them here, is that correct?

Reply:
Oh sorry I missed the change from `agent` to `agentClass`.

Reply:
@whikloj  Yes, I am filtering out FOAF_AGENT_VALUE from the agents list and allowing only FOAF_AGENT_VALUE in the agentClasses list of the auth.